### PR TITLE
Add max and min math functions

### LINKED ISSUE #7856: Add max and min math functions
It's not super common, but I've found myself wanting these functions to set upper and lower bounds on things.

Obviously this is just a nice tool to have, but as a specific use case, some `string` commands like `pad` and `repeat` will error if you give them negative numbers. Obviously we can't use `abs` here, so I've been doing `test $blah -lt 0 && set $blah 0`. This works of course, but it is less elegant and requires setting the value to a variable first.

### PR DESCRIPTION
**Pull request by IlanCosman**
## Description

Adds `max` and `min` math functions.

fixes #7856 

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

**Comment by IlanCosman** [comment]
@ridiculousfish Could you review? Fabian is on break and I'd like to spare him the trouble of fixing my terrible C++ 😄

**Comment by IlanCosman** [comment]
Flake test I think?

File: src/tinyexpr.cpp [hunk 1]
```diff
@@ -182,6 +182,10 @@ static constexpr double bit_xor(double a, double b) {
     return static_cast<double>(static_cast<long long>(a) ^ static_cast<long long>(b));
 }
 
+static double max(double a, double b) { return std::max(a,b); }
+
+static double min(double a, double b) { return std::min(a,b); }
```
Comment (krobelus) [inline src/tinyexpr.cpp:187]: `clang-format` should add a space between arguments :detective:

**Comment by krobelus** [comment]
> Flake test I think?

Yeah, sorry - I've disabled the test in 4f00ccfd1

File: src/tinyexpr.cpp (hunk 1 above)
Comment (IlanCosman) [reply src/tinyexpr.cpp]: My bad, I don't actually have `clang` on my machine 😄

**Comment by ridiculousfish** [comment]
Are we OK with `min` and `max` which only support two arguments?  I suppose we can always extend it later.

Regarding the implementations, there's some floating point edge-cases, [we should just match Java](https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#max-double-double-).

1. Negative zero should be "less than" positive zero. So `math 'max(-0, 0)'` should print 0, and min should print -0.
2. `max` and `min` of anything with NaN should be NaN. `math 'max(1, 0/0)'` should error instead of printing 1.

Something like this should do it:

   double max(double a, double b) {
       if (std::isnan(a)) return a;
       if (std::isnan(b)) return b;
       if (a == b) return std::signbit(a) ? b : a; // treat +0 as larger than -0
       return a > b ? a : b;
   }

   double min(double a, double b) {
       if (std::isnan(a)) return a;
       if (std::isnan(b)) return b;
       if (a == b) return std::signbit(a) ? a : b; // treat -0 as smaller than +0
       return a < b ? a : b;
   }

**Comment by IlanCosman** [comment]
Thanks @ridiculousfish! Yah I decided that two arguments is better than none at all, and in many (most?) cases two is all that is necessary.

Thanks for the changes, I'll add them 😄 Is there any reason that C++'s `max` and `min` don't do those things already?

**Comment by ridiculousfish** [comment]
`std::max` and `std::min` are generic functions, they operate on anything comparable so they are ignorant of floating point's nonsense rules.

**Comment by ridiculousfish** [comment]
Merged as c762c624647f6c188177c79f62608fb844f2ba8f with some fixes to Mac build. Thank you!!

**Pull request closed without merging**